snippets/omt.snippets.json

```diff
@@ -0,0 +1,98 @@
+{
+    // Place your snippets for odt here. Each snippet is defined under a snippet name and has a prefix, body and
+    // description. The prefix is what is used to trigger the snippet and the body will be expanded and inserted. Possible variables are:
+    // $1, $2 for tab stops, $0 for the final cursor position, and ${1:label}, ${2:another} for placeholders. Placeholders with the
+    // same ids are connected.
+    // Example:
+    // "Print to console": {
+    //     "prefix": "log",
+    //     "description": "Log output to console",
+    //     "body": [
+    //         "console.log('$1');",
+    //         "$2"
+    //     ]
+    // }
+
+    // MODULE ENTRY POINT / DOCUMENT CONSTRUCTS
+    "Menu definition": {
+        "prefix": "menu",
+        "description": "Define sections of the application menu.\n\nUse in OMT modules only.",
+        "body": [
+            "menu:",
+            "    ${1:SectionName}:",
+            "    -   title: ${2:Menu item name}",
+            "        onSelect: |",
+            "            ${0:@LOG('snippet');}"
+        ]
+    },
+    "Global actions": {
+        "prefix": "global-actions",
+        "description": "Define global actions.\n\nUse in OMT modules only.",
+        "body": [
+            "actions:",
+            "    ${1|dashboard,dossier,notifications,dashboardActions|}:",
+            "    -    id: $2",
+            "         title: ${3:Action name}",
+            "         icon: ${4:Action icon}",
+            "         onSelect: |",
+            "             ${0:@LOG('snippet');}"
+        ]
+    },
+    "Declared imports": {
+        "prefix": "declare",
+        "description": "Declare modules with their imports.\n\nUse in OMT modules only.",
+        "body": [
+            "declare:",
+            "    ${1:ModuleName}:",
+            "        ${2:ThingToDeclare}:",
+            "            type: ${3|Activity,Command,Procedure|}",
+            "            params:",
+            "            -   ${4:primitive type or prefixed IRI}",
+            "            returns: ${0:primitive type or prefixed IRI}"
+        ]
+    },
+    "Imports": {
+        "prefix": "import",
+        "description": "Import definitions from another OMT file and/or module.",
+        "body": [
+            "import:",
+            "    ${1|./,module:|}:",
+            "    -   ${0:ThingToImport}"
+        ]
+    },
+
+    // MODEL TEMPLATES
+    "Activity model": {
+        "prefix": "model-act",
```

Comment:
Bijzondere reden om hier niet gewoon "activity" te kiezen? (en zelfde voor procedure hieronder)

Reply:
Ja, op het moment dat je zelf al de definitie aan het typen bent, type je ook activity. Daarna is je regel afgelopen en druk je dus op enter:

```yaml
...
model:
    MyActivity: !Activity
```

Wanneer we als key voor een activity "activity" hebben, wordt nu het template ingevoegd en krijg je zoiets:

```yaml
...
model:
    MyActivity: !ActivityName: !Activity
        title: Display name of the activity
        ...
```

Door de prefix op "model-act" te houden, wordt de snippet niet ingevoegd als je "Activity" uittypt.